preferences/models.py

```diff
@@ -1,3 +1,41 @@
 from django.db import models
+from django.contrib.auth.models import User
+from users.models import AppUser
 
-# Create your models here.
+from django.utils.translation import gettext_lazy as _
+
+# Model for the database representation of a teacher Preferences record.
+class Preferences(models.Model):
+
+    class SabbaticalLength(models.TextChoices):
+        HALF_LENGTH = 'HALF', _('Half Length')
+        FULL_LENGTH = 'FULL', _('Full Length')
+        NONE = 'NONE', _('None')
+
+    professor = models.OneToOneField(
+        AppUser,
+        on_delete=models.CASCADE,
+        to_field='user',
+        primary_key=True
+    )
+    is_submitted = models.BooleanField(default=False)
+    unavailable_sem1 = models.CharField(max_length=7, blank=True, default='')   #'YYYY-MM'
```

Comment:
Change unavailable_sem1 and unavailable_sem2 to align courses semester offering boolean expression.

Reply:
Fields are changed to Boolean types and renamed.